kotest-assertions-arrow bug in shouldRaise

**Which version of Kotest are you using**
Kotest: 6.1.2
kotest-assertions-arrow: 2.0.0

I am using `shouldRaise { ... }` in my project and when there is a problem with this assertion, there seems to be a missing FailureKt class. Instead of showing the correct error message, you get this Stacktrace:

```
java.lang.NoClassDefFoundError: io/kotest/assertions/FailuresKt

    Caused by: java.lang.NoClassDefFoundError: io/kotest/assertions/FailuresKt

    Caused by: java.lang.ClassNotFoundException: io.kotest.assertions.FailuresKt
```

I duplicated the function and replaced the code `throw failure("...")` in it with a simple `fail("...")` which fixes the problem.

But I got another issue with the way this function is conceptionally written:
When you simply want to assert that "something raises an error", and you do not care about the error type, you then would code `shouldRaise { raise "Boom" }` - but this does not succeed, because the type inference would infer `Unit` as the expected type for E. Which is wrong...

--- a/documentation/docs/assertions/arrow.md
+++ b/documentation/docs/assertions/arrow.md
@@ -5,13 +5,11 @@ sidebar_label: Arrow
 ---
 
 
-[![Latest Release](https://img.shields.io/maven-central/v/io.kotest.extensions/kotest-assertions-arrow)](https://search.maven.org/artifact/io.kotest.extensions/kotest-assertions-arrow)
-
 This page lists all current matchers in the Kotest arrow matchers extension library.
 
 :::note
-The following module is needed: `io.kotest.extensions:kotest-assertions-arrow` which is versioned independently of the main Kotest project.
-Search maven central for latest version [here](https://central.sonatype.com/search?q=io.kotest.extensions:kotest-assertions-arrow).
+The `kotest-assertions-arrow` module has moved to the `io.kotest` group. Please update your dependency to
+`io.kotest:kotest-assertions-arrow`, and use the version matching your kotest version.
 :::
 
 :::note

--- a/documentation/versioned_docs/version-6.0/assertions/arrow.md
+++ b/documentation/versioned_docs/version-6.0/assertions/arrow.md
@@ -4,14 +4,11 @@ slug: arrow.html
 sidebar_label: Arrow
 ---
 
-
-[![Latest Release](https://img.shields.io/maven-central/v/io.kotest.extensions/kotest-assertions-arrow)](https://search.maven.org/artifact/io.kotest.extensions/kotest-assertions-arrow)
-
 This page lists all current matchers in the Kotest arrow matchers extension library.
 
 :::note
-The following module is needed: `io.kotest.extensions:kotest-assertions-arrow` which is versioned independently of the main Kotest project.
-Search maven central for latest version [here](https://central.sonatype.com/search?q=io.kotest.extensions:kotest-assertions-arrow).
+The `kotest-assertions-arrow` module has moved to the `io.kotest` group. Please update your dependency to
+`io.kotest:kotest-assertions-arrow`, and use the version matching your kotest version.
 :::
 
 :::note

--- a/documentation/versioned_docs/version-6.1/assertions/arrow.md
+++ b/documentation/versioned_docs/version-6.1/assertions/arrow.md
@@ -4,14 +4,11 @@ slug: arrow.html
 sidebar_label: Arrow
 ---
 
-
-[![Latest Release](https://img.shields.io/maven-central/v/io.kotest.extensions/kotest-assertions-arrow)](https://search.maven.org/artifact/io.kotest.extensions/kotest-assertions-arrow)
-
 This page lists all current matchers in the Kotest arrow matchers extension library.
 
 :::note
-The following module is needed: `io.kotest.extensions:kotest-assertions-arrow` which is versioned independently of the main Kotest project.
-Search maven central for latest version [here](https://central.sonatype.com/search?q=io.kotest.extensions:kotest-assertions-arrow).
+The `kotest-assertions-arrow` module has moved to the `io.kotest` group. Please update your dependency to
+`io.kotest:kotest-assertions-arrow`, and use the version matching your kotest version.
 :::
 
 :::note

